src/par_iter/string.rs

```diff
@@ -68,3 +81,103 @@ impl<'a> IntoIterator for ParChars<'a> {
         self.chars.chars()
     }
 }
+
+
+// /////////////////////////////////////////////////////////////////////////
+
+pub struct ParSplit<'a> {
+    chars: &'a str,
+    separator: char,
+
+    /// Keeps track of the first separator found in the string.  This lets us
+    /// quickly answer `can_split`, and it also corresponds to what parts we've
+    /// already scanned as we keep splitting smaller.
+    first: Option<usize>,
+}
+
+impl<'a> ParSplit<'a> {
+    fn new(chars: &'a str, separator: char) -> Self {
+        ParSplit {
+            chars: chars,
+            separator: separator,
+            first: chars.find(separator),
+        }
+    }
+}
+
+impl<'a> ParallelIterator for ParSplit<'a> {
+    type Item = &'a str;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        bridge_unindexed(self, consumer)
+    }
+}
+
+impl<'a> UnindexedProducer for ParSplit<'a> {
+    fn can_split(&self) -> bool {
+        self.first.is_some()
+    }
+
+    fn split(self) -> (Self, Self) {
+        let ParSplit { chars, separator, first } = self;
+        let first = first.expect("error: splitting with no separator present!");
+
+        // First find a suitable UTF-8 boundary in the unsearched region.
+        let char_index = find_char_midpoint(&chars[first..]) + first;
+
+        // Find a separator in reverse, towards the `first` that we know exists.
+        let index = chars[first..char_index]
```

Comment:
I started down this road, using `Cell` for `can_split` updates, but otherwise basically what you suggest.  It was easy if I didn't care about re-scanning memory we've already covered, but I was getting tripped up trying to avoid that.  Especially, scanning both forward and backwards meant there were two "already scanned" indexes to maintain.

It also doesn't help that `can_split` is called before `Splitter::try`.  Either way we order it, one of those will be doing work that the other might nullify, which is why I tried to keep `can_split` minimal.  But I'm prescanning `first`, so maybe that's moot.  Hrm.  At least `Splitter::try` is well-bounded, so maybe we _should_ do that first.

I'll play more and see if I can make this a bit lazier...